src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs

```diff
@@ -253,23 +254,22 @@ protected override SyntaxTrivia TogglePragmaDirective(SyntaxTrivia trivia)
             return SyntaxFactory.Trivia(newPragmaWarning);
         }
 
-        protected override SyntaxToken GetAdjustedTokenForPragmaRestore(
-            SyntaxToken token, SyntaxNode root, TextLineCollection lines, int indexOfLine)
+        protected override SyntaxToken GetAdjustedTokenForPragmaDisable(SyntaxToken token, SyntaxNode root, TextLineCollection lines, int indexOfLine)
         {
-            var nextToken = token.GetNextToken();
-            if (nextToken.Kind() == SyntaxKind.SemicolonToken &&
-                nextToken.Parent is StatementSyntax statement &&
-                statement.GetLastToken() == nextToken &&
-                token.Parent.FirstAncestorOrSelf<StatementSyntax>() == statement)
+            // normally we find the parent statement for the token but if the token is a close brace
+            // that could expand to a very large block, so we skip it and just return the token itself
+            if (token.IsKind(SyntaxKind.CloseBraceToken))
```

Comment:
This is for:

```
{
   // some arbitrary code
#pragma this_is_an_invalid_pragma
}
```

Without this, the disable and restore would be around the whole block, so I added this. Not sure if its worth it?

VB doesn't have arbitrary `#pragma`s so its not an issue there as far as I can tell.

Reply:
Seems like the right thing to me

Reply:
i don't think we need this.  indeed, for a block, i like the idea that we'd wrap the entire block.